### PR TITLE
Add Support for PHP among Highlight SDKs

### PR DESCRIPTION
**Pull request by ayewo**
## Summary

The PHP SDK closely follows the design of Java SDK mentioned in the original issue: https://github.com/highlight/highlight/issues/4225

## How did you test this change?

Automated tests.

## Are there any deployment considerations?

Requires a minimum of PHP v8.2+ as some Otel dependencies wont work with older versions like PHP v7.4.

## Does this work require review from our design team?

Nope.

/claim #4225

### CODE REVIEW

**Comment by ayewo** [comment]
@Vadman97 @jay-khatri  this is ready for review.

File: sdk/highlight-php/src/SDK/HighlightOpenTelemetry.php [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        $spanProcessor = new BatchSpanProcessor($exporter, ClockFactory::getDefault());
+        $sampler = new AlwaysOnSampler();
+
+        $this->tracerProvider = (new TracerProviderBuilder())
+            ->addSpanProcessor($spanProcessor)
+            ->setResource($resource)->setSampler($sampler);
+
+
+        // Log
+        $transport = (new OtlpHttpTransportFactory())->create(HighlightRoute::buildLogRoute($options->getBackendUrl()), 'application/json');
```
Comment (Vadman97) [inline sdk/highlight-php/src/SDK/HighlightOpenTelemetry.php:109]: can we put `application/json` into a class attribute similar to `self::$compression`?

File: sdk/highlight-php/src/SDK/HighlightOpenTelemetry.php [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        $exporter = new LogsExporter($transport);
+        $logProcessor = new BatchLogRecordProcessor($exporter, ClockFactory::getDefault());
```
Comment (Vadman97) [inline sdk/highlight-php/src/SDK/HighlightOpenTelemetry.php:111]: does the optenelemetry php sdk allow configuring batching settings? we have some customization there on the java sdk

File: sdk/highlight-php/tests/Unit/SDK/Common/HighlightOptionsBuilderTest.php
Comment (Vadman97) [inline sdk/highlight-php/tests/Unit/SDK/Common/HighlightOptionsBuilderTest.php:1]: how would someone run the tests? could we add github actions under `.github/workflows` to run PHP tests in CI? could we also document what it takes to run the tests in a `README.md` file?

**Review by Vadman97 — COMMENTED** [review]
thank you for your contribution @ayewo! looks great overall; just asking a few clarifying questions.

could you also add documentation for setting up the SDK similar to https://www.highlight.io/docs/getting-started/backend-sdk/java/other and https://www.highlight.io/docs/sdk/java?

File: sdk/highlight-php/src/SDK/HighlightOpenTelemetry.php (hunk 1 above)
Comment (ayewo) [reply sdk/highlight-php/src/SDK/HighlightOpenTelemetry.php]: Sure. Added them as:
```php
    private static string $format = 'application/json';
    private static array $headers = [];
    private static string $compression = 'gzip';
    private static float $timeout = 30.;
```

File: sdk/highlight-php/src/SDK/HighlightOpenTelemetry.php (hunk 2 above)
Comment (ayewo) [reply sdk/highlight-php/src/SDK/HighlightOpenTelemetry.php]: I've updated the batch settings to match the values used by the Java SDK.

File: sdk/highlight-php/tests/Unit/SDK/Common/HighlightOptionsBuilderTest.php
Comment (ayewo) [reply sdk/highlight-php/tests/Unit/SDK/Common/HighlightOptionsBuilderTest.php:1]: 1. Added a `README.md` to the PR;
2. Running the tests is pretty straight forward:
```php
 ./vendor/bin/phpunit --verbose tests
```

You'll need to setup a development environment for the PHP dependencies & extensions (perhaps in Docker).
But adding a GitHub Actions `workflow.yml` is very much out of scope of this already massive PR ...

**Comment by ayewo** [comment]
@Vadman97 I’ve addressed all the feedback raised in your review.

**Comment by Vadman97** [comment]
@ayewo could you run `yarn format:all` to lint the new code?

**Comment by ayewo** [comment]
> could you run `yarn format:all` to lint the new code?

@Vadman97 done.

**Comment by Vadman97** [comment]
Thanks @ayewo ! Taking a close look again at this today. Meanwhile, do you have any ideas for how we would publish the SDK for our customers to use? Do we upload it to some artifacts repository like with java / npm / pip ? Would like to see the functional tests + the artifact release happening as part of the CI.

**Comment by Vadman97** [comment]
ah, looks like we [release tags](https://medium.com/@mazraara/create-a-composer-package-and-publish-3683596dec45) and packagist will pick them up

**Comment by ayewo** [comment]
> Thanks @ayewo ! Taking a close look again at this today. 

@Vadman97 Appreciate that. I would like to have this wrapped up as soon as possible.

> Meanwhile, do you have any ideas for how we would publish the SDK for our customers to use? Do we upload it to some artifacts repository like with java / npm / pip ? Would like to see the functional tests + the artifact release happening as part of the CI.

With respect to the `yarn` CLI and the [npm](https://www.npmjs.com/) registry, PHP's equivalents are [`composer`](https://getcomposer.org/) and [Packagist](https://packagist.org/).

So, cutting a new release for the PHP SDK will be similar to how you already do it for other languages using GH Actions.

All the concerns you've raised so far: CI, packaging/release etc are operational concerns so I think it is best to create separate issue + bounty to discuss and implement them rather than draw out the review of this PR.

**Review by Vadman97 — APPROVED** [review]
thank you for your contribution @ayewo :tada:

**Comment by ayewo** [comment]
@Vadman97 thank you 🙏🏻!

Once you have the new issue drafted for the CI and release stuff, feel free to tag me here or on Discord.

**Comment by matthiscock** [comment]
ITs great that Laravel has been integrated - but cant see any docs on it
How do we go about using this

**Pull request merged**